content/en/user-guide/integrations/aws-sam/index.md

```diff
@@ -5,35 +5,87 @@ description: >
+
+### Deploy the SAM application
+
+After initializing the SAM project, enter the project directory and deploy the application using the following command:
+
+{{< command >}}
+$ samlocal deploy --guided
+{{< / command >}}
+
+Enter the default values for the deployment, such as the stack name, region, and confirm the changes. The `samlocal` wrapper will package and deploy the application to LocalStack.
 
 ### Configuration
 
-* `AWS_ENDPOINT_URL`: The endpoint URL (i.e., protocol, host, and port) to connect to LocalStack (default: `http://localhost:4566`)
+| Environment Variable   | Default value                                    | Description                                                             |
+|------------------------|--------------------------------------------------|-------------------------------------------------------------------------|
+| AWS_ENDPOINT_URL       | `http://localhost.localstack.cloud:4566`        | URL at which the `boto3` client can reach LocalStack                   |
+| EDGE_PORT              | `4566` (Deprecated)                              | Port number under which the LocalStack edge service is available        |
+| LOCALSTACK_HOSTNAME    | `localhost` (Deprecated)                         | Host under which the LocalStack edge service is available
+
+## Debugging on VS Code
+
+To debug your Lambda functions in VS Code while using the SAM CLI with LocalStack, set up a launch configuration in the `.vscode/launch.json` file. Insert the following settings into the file:
+
+
+```json
+{
+            "type": "aws-sam",
+            "request": "direct-invoke",
+            "name": "Job dispatcher lambda",
+            "invokeTarget": {
+                "target": "code",
+                "projectRoot": "${workspaceFolder}",
+                "lambdaHandler": "lambda/lambda.handler"
+            },
+            "lambda": {
+                "runtime": "python3.8",
+                "payload": {},
+                "environmentVariables": {
+                    "ENDPOINT_URL": "http://localstack:4566/",
+                    "S3_ENDPOINT_URL": "http://s3.localhost.localstack.cloud:4566/",
+                    "AWS_ACCESS_KEY_ID": "test",
+                    "AWS_SECRET_ACCESS_KEY": "test",
+                    "AWS_SESSION_TOKEN": "test",
+                    "AWS_REGION": "us-east-1",
+                    "MAIN_DOCKER_NETWORK": "localstack-network"
+                }
+            },
+            "sam": {
+                "dockerNetwork": "localstack-network"
+            }
+        }
+```
+
+The `dockerNetwork` property is essential as it allows the LocalStack container to use the `samlocal invoke` commands within the same network as the LocalStack container itself. Adjust the Lambda function handler and environment variables as needed.
```

Comment:
The mentioned issue (#1234 ) is about `sam local` and NOT `samlocal`. By default the container is span up in the main docker network and accessible, however IF the user decides to run LocalStack on a different network ONLY then they need to modify this.

Reply:
Here is the communication from the reporter, [channel](https://localstack-community.slack.com/archives/C03F7EQM2M8/p1686305656429799).

Reply:
He luckily shared the actual LS startup command and it seems he puts his LS instance deliberately to a different network.
![image](https://github.com/localstack/docs/assets/43646571/0731099c-fa62-419a-b0f2-26f9f157be8f)
Regarding the question if he uses `samlocal local` or `sam local` is unclear, but doesn't really matter as he's setting the env vars for the debug session in the shared config.